File: Exercises2024/Chapter06_vle_free_energy/Fortran/1-Gibbs/Source/toterg.f
      Subroutine Toterg(Ener, Vir, BoxID)

C     ---Calculates Total Energy

      Implicit None

      Include 'parameter.inc'
      Include 'conf.inc'
       
      Double Precision Xi, Yi, Zi, Ener, Eni, Viri, Vir
      Integer I, Jb, BoxID
 
      Ener = 0.0d0
      Vir = 0.0d0
      Do I = 1, Npart - 1
         If (Id(I).Eq.BoxID) Then
            Xi = X(I)
            Yi = Y(I)
            Zi = Z(I)
            Jb = I + 1
            Call Eneri(Xi, Yi, Zi, I, Jb, Eni, Viri, BoxID)
            Ener = Ener + Eni
            Vir = Vir + Viri
         End If
      End Do

      Return
      End

File: Exercises2024/Chapter06_vle_free_energy/C/1-Gibbs/Source/toterg.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>
#include "system.h"
#include "conf.h"

void Toterg(double *Ener,double *Vir,int BoxID)
{
  //     ---Calculates Total Energy
  
  double Xi, Yi, Zi, Eni, Viri;
  int I, Jb;
  
  *Ener = 0.0;
  *Vir = 0.0;
  for(I = 0;I<Npart-1;I++)
    {
      if(Id[I]==BoxID) 
	{
	  Xi = X[I];
	  Yi = Y[I];
	  Zi = Z[I];
	  Jb = I + 1;
	  Eneri(Xi, Yi, Zi, I, Jb, &Eni,&Viri, BoxID);
	  *Ener = *Ener + Eni;
	  *Vir = *Vir + Viri;
	}
    }
  return;
}
